lib/archivist/vaults/redis/index.js

```diff
@@ -92,6 +92,31 @@ RedisVault.prototype._unprefix = function (key) {
 };
 */
 
+RedisVault.prototype.scan = function (key, cb) {
+	var set = new Set()
+	key = this._prefix(key);
+
+	this.logger.error('scan:', key);
+
+	const rscan = (cursor) => {
+		this.client.scan([cursor, 'MATCH', key, 'COUNT', 100], function (error, data) {
+			if (error) {
+				return cb(error);
+			}
+
+			data[1].forEach((k) => set.add(k.toString()));
```

Comment:
Why would we expect duplicates? If not, a simple array would do I think.

Reply:
yes, SCAN could return duplicates on a full iteration. that's a drawback of the command